pkg/internal/vault/vault.go

```diff
@@ -366,3 +374,17 @@ func extractCertificatesFromVaultCertificateSecret(secret *certutil.Secret) ([]b
 
 	return []byte(strings.Join(crtPems, "\n")), caPem, nil
 }
+
+func (v *Vault) IsVaultInitializedAndUnsealed() error {
+	healthURL := path.Join("/v1", "sys", "health")
```

Comment:
Not a fan of doing a raw request here, really no way to check this with the library? if not could we file a feature request to add this

Reply:
We have a library function that we can call on the vault package, but there an issue with the said function. The problem is that when token passed to the vault does not have sufficient permissions to call the `/sys/*` endpoints, vault returns the correct HTTP status code, but an empty JSON response. Vault client's `client.Sys().Health()`, is replying purely on the JSON response of the vault and not the HTTP status code by setting the following parameters:
```golang
r.Params.Add("uninitcode", "299")
r.Params.Add("sealedcode", "299")
r.Params.Add("standbycode", "299")
r.Params.Add("drsecondarycode", "299")
r.Params.Add("performancestandbycode", "299")
```
This is why we need to make a raw request here. 
Honestly speaking, I couldn't think of a another approach to fix this issue. Please feel free to suggest a better approach and I would be more than happy to implement the same. :)